ogr/ogrsf_frmts/ogrsf_frmts.h

```diff
@@ -414,15 +414,17 @@ class CPL_DLL OGRLayer : public GDALMajorObject
     {
 
         FieldDefnChange(std::unique_ptr<T> &&poFieldDefnIn, int iFieldIn,
-                        FieldChangeType eChangeTypeIn)
+                        FieldChangeType eChangeTypeIn, bool bGeneratedIn)
             : poFieldDefn(std::move(poFieldDefnIn)), iField(iFieldIn),
-              eChangeType(eChangeTypeIn)
+              eChangeType(eChangeTypeIn), bGenerated(bGeneratedIn)
         {
         }
 
         std::unique_ptr<T> poFieldDefn;
         int iField;
         FieldChangeType eChangeType;
+        // Used by drivers (GPKG) to track generated fields
+        bool bGenerated;
```

Comment:
Just thinking that it could be perhaps worthwhile to have the generated boolean in OGRFieldDefn itself (could apply to other drivers), but that's likely out of scope of this PR / work ?

Reply:
I agree it could be better moved to OGRFieldDefn but I'd prefer to do it in a separate PR.

Reply:
fine. I'm merging this as it